sympy/core/tests/test_arit.py

```diff
@@ -1913,6 +1913,20 @@ def test_Mod():
     assert Mod(x, y).rewrite(floor) == x - y*floor(x/y)
     assert ((x - Mod(x, y))/y).rewrite(floor) == floor(x/y)
 
+    # issue 21373
+    from sympy.core.cache import clear_cache
+    from sympy.functions.elementary.trigonometric import sinh
+    from sympy.functions.elementary.piecewise import Piecewise
+
+    x_r, y_r = symbols('x_r y_r', real=True)
+    clear_cache()
+    (Piecewise((x_r, y_r > x_r), (y_r, True)) / z) % 1
```

Comment:
I don't think it's necessary to clear the cache here. The issue shows up reliably first time. The cache just prevents it from showing up second time in some cases.

Reply:
You're right - I hadn't realized it wasn't all cases. The latest push should resolve this and the extra formatting from my editor.